### ISSUE TITLE
incorrect tree shaking with `import * as foo` + `export default foo`

### ISSUE THREAD

**Issue by sapphi-red** [rollup/rollup#4751]
### Rollup Version

3.7.4

### Operating System (or Browser)

stackblitz

### Link To Reproduction

https://stackblitz.com/edit/rollup-template-vg2pmh?file=src%2Fmain.js

### Expected Behaviour

Running the bundled file output
```
hello from greetings
hi from greetings
```
as before bundled.

### Actual Behaviour

Running the bundled file output 
```
hi from greetings
```
.

`api.greetings.hello();` is removed even if it has a side effect.

It works when `src/api/index.js` is changed like below
```diff
export default {
-  greetings,
+  greetings: {
+    hello: greetings.hello,
+  },
};
```

Original issue: https://github.com/vitejs/vite/issues/11356

**Comment by sapphi-red** [comment]
This doesn't happen with 2.79.1 and happens with 3.0.0.

**Comment by lukastaegert** [comment]
As it turns out, the case where a namespace is itself nested inside an object was not yet handled correctly. Fix for this and similar issues at #4758.

**Comment by sapphi-red** [comment]
Thanks!
